sdk/servicebus/service-bus/src/receiver.ts

```diff
@@ -523,8 +534,12 @@ export class SessionReceiver {
       throw new Error("'onError' is a required parameter and must be of type 'function'.");
     }
     this._createMessageSessionIfDoesntExist()
-      .then(() => {
-        this._messageSession!.receive(onMessage, onError, options);
+      .then(async () => {
+        if (!this._isClosed && this._messageSession) {
+          this._messageSession.receive(onMessage, onError, options);
+        } else {
+          await this._messageSession!.close();
```

Comment:
We will enter the `else` block when `this._messageSession` is undefined/null as well. I would suggest an early exit using
```
if (!this._messageSession) {
    return;
}
```